localstack/services/sns/resource_providers/aws_sns_topic.py

```diff
@@ -0,0 +1,178 @@
+
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+from localstack.utils.strings import short_uid
+
+
+class SNSTopicProperties(TypedDict):
+    ContentBasedDeduplication: Optional[bool]
+    DataProtectionPolicy: Optional[dict]
+    DisplayName: Optional[str]
+    FifoTopic: Optional[bool]
+    KmsMasterKeyId: Optional[str]
+    SignatureVersion: Optional[str]
+    Subscription: Optional[list[Subscription]]
+    Tags: Optional[list[Tag]]
+    TopicArn: Optional[str]
+    TopicName: Optional[str]
+    TracingConfig: Optional[str]
+
+
+class Subscription(TypedDict):
+    Endpoint: Optional[str]
+    Protocol: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class SNSTopicProvider(ResourceProvider[SNSTopicProperties]):
+    TYPE = "AWS::SNS::Topic"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[SNSTopicProperties],
+    ) -> ProgressEvent[SNSTopicProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/TopicArn
+
+
+
+        Create-only properties:
+          - /properties/TopicName
+          - /properties/FifoTopic
+
+        Read-only properties:
+          - /properties/TopicArn
+
+        IAM permissions required:
+          - sns:CreateTopic
+          - sns:TagResource
+          - sns:Subscribe
+          - sns:GetTopicAttributes
+          - sns:PutDataProtectionPolicy
+
+        """
+        model = request.desired_state
+        sns = request.aws_client_factory.sns
+        # TODO: validations and iam checks
+
+        attributes = {k: v for k, v in model.items() if v is not None if k != "TopicName"}
+
+        # following attributes need to be str instead of bool for boto to work
+        if attributes.get("FifoTopic") is not None:
+            attributes["FifoTopic"] = str(attributes.get("FifoTopic"))
+
+        if attributes.get("ContentBasedDeduplication") is not None:
+            attributes["ContentBasedDeduplication"] = str(
+                attributes.get("ContentBasedDeduplication")
+            )
+
+        subscriptions = []
+        if attributes.get("Subscription") is not None:
+            subscriptions = attributes["Subscription"]
+            del attributes["Subscription"]
+
+        # in case cloudformation didn't provide topic name
+        if model.get("TopicName") is None:
+            model["TopicName"] = f"topic-{short_uid()}"
+
+        create_sns_response = sns.create_topic(Name=model["TopicName"], Attributes=attributes)
```

Comment:
The attributes still potentially contain the raw `Tags` from the properties, which isn't valid as an attribute value unfortunately (see CI failure [here](https://app.circleci.com/pipelines/github/localstack/localstack/18238/workflows/1c229624-f507-48b6-95dd-11bb935fbe67/jobs/140322) )
```
...
  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/botocore/validate.py", line 381, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter Attributes.Tags, value: [{'Key': 'param-value', 'Value': 'param ls-param-value-fdccf91f'}], type: <class 'list'>, valid types: <class 'str'>
```

Reply:
I can modify it so tags are extracted from attributes into a list. 
Then I would delete them from attributes and apply them separately, similarly to how I did it with subscriptions.

It's a bit odd, that same as for subscription, there is a valid string value to which we should be able to convert it (at least according to API), but everything I tried didn't produce any results.

Reply:
Done